com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/json/references/JsonDocumentManager.java

```diff
@@ -112,9 +112,12 @@ private JsonNode parse(URL url) {
         }
     }
 
+    /*
+     * Returns a copy of the URL parameter without the fragment or query.
+     */
```

Comment:
Misleading comment. Only the fragment is omitted. `URL#getFile()` returns the path portion plus the query portion, if it exists. Not using the query would be an error, since query strings are often required to properly address documents.

Reply:
Might also be worth converting to URI and normalizing the path, and using that as a key for loaded docs. `URL#equals` does hostname resolution, so using it as a map key will do so as well. It's therefore not generally recommended as a map key or set element.

The result isn't perfect. URL.equals does its host resolution to recognize when two host names really refer to the same host, and URI doesn't do that. Also, rearrangements of query params are generally meaningless, so it'd be nice to sort them or something; but that's not guaranteed, so you can't do that. Unfortunately, URL/URI identity detection is not easy. But requiring hostname resolution in map operations is probably best avoided.